workflow-testing/src/main/java/com/squareup/workflow/testing/RenderTester.kt

```diff
@@ -296,19 +343,42 @@ interface RenderTester<PropsT, StateT, OutputT, RenderingT> {
    * @param output If non-null, [WorkflowOutput.value] will be "emitted" when this workflow is
    * rendered. The [WorkflowAction] used to handle this output can be verified using methods on
    * [RenderTestResult].
+   * @param description Optional string that will be used to describe this expectation in error
+   * messages.
    */
   @OptIn(ExperimentalWorkflowApi::class)
   fun <ChildPropsT, ChildOutputT, ChildRenderingT> expectWorkflow(
     workflowType: KClass<out Workflow<ChildPropsT, ChildOutputT, ChildRenderingT>>,
     rendering: ChildRenderingT,
     key: String = "",
     assertProps: (props: ChildPropsT) -> Unit = {},
-    output: WorkflowOutput<ChildOutputT>? = null
+    output: WorkflowOutput<ChildOutputT>? = null,
+    description: String = ""
   ): RenderTester<PropsT, StateT, OutputT, RenderingT> =
-    expectWorkflow(workflowType.workflowIdentifier, rendering, key, output = output, assertProps = {
-      @Suppress("UNCHECKED_CAST")
-      assertProps(it as ChildPropsT)
-    })
+    expectWorkflow(
+        workflowType.workflowIdentifier, rendering, key, output = output, description = description,
+        assertProps = {
+          @Suppress("UNCHECKED_CAST")
+          assertProps(it as ChildPropsT)
+        })
+
+  /**
+   * Specifies that this render pass is expected to render a particular child workflow.
+   *
+   * @param description String that will be used to describe this expectation in error messages.
+   * @param exactMatch If true, then the test will fail if any other matching expectations are also
+   * exact matches, and the expectation will only be allowed to match a single child workflow.
+   * If false, the match will only be used if no other expectations return exclusive matches (in
+   * which case the first match will be used), and the expectation may match multiple children.
+   * @param matcher A function that determines whether a given [RenderChildInvocation] matches this
+   * expectation by returning a [ChildWorkflowMatch]. If the expectation matches, the function
+   * must include the rendering and optional output for the child workflow.
+   */
+  fun expectWorkflow(
+    description: String,
```

Comment:
`description` is required for this overload intentionally, since we can't derive a human-readable description from a predicate function.

Reply:
Added a note about this in the kdoc.